### ISSUE TITLE
Zettlr v3.0.0-beta.3 has FSAL Cache bug

### ISSUE THREAD

**Issue by nathanlesage** [Zettlr/Zettlr#4269]
### Description

I noticed that after I've upgraded my own install to v3.0.0-beta.3 (but locally compiled nevertheless), I noticed that the startup times were very high (> 10sec), indicating that there were a ton of cache misses happening under the hood.

Investigating, I found that somehow the cache returns bogus values. Specifically, the actual cached descriptors are in string-form, instead of actually being in a revitalized object form. See the following example I got from running `JSON.parse(content)` on one of the shards:

```json
[
  [
    '/Users/hendrik/Documents/Zettlr Tutorial/citing.md',
    '{"root":false,"dir":"/Users/hendrik/Documents/Zettlr Tutorial","path":"/Users/hendrik/Documents/Zettlr Tutorial/citing.md","name":"citing.md","ext":".md","size":5257,"id":"","tags":[],"links":[],"bom":"","type":"file","wordCount":719,"charCount":3873,"modtime":1679442158538,"creationtime":1679442158538,"linefeed":"\\n","firstHeading":"Citing with Zettlr 💬","yamlTitle":"Citing with Zettlr","frontmatter":{"title":"Citing with Zettlr","keywords":["Zotero","JabRef","CSL JSON","BibTex","Reference Management"]},"modified":false}'
  ]
]
[
  [
    '/Users/hendrik/Documents/Zettlr Tutorial/split-view-intro.md',
    '{"root":false,"dir":"/Users/hendrik/Documents/Zettlr Tutorial","path":"/Users/hendrik/Documents/Zettlr Tutorial/split-view-intro.md","name":"split-view-intro.md","ext":".md","size":855,"id":"","tags":[],"links":[],"bom":"","type":"file","wordCount":149,"charCount":672,"modtime":1679442158540,"creationtime":1679442158540,"linefeed":"\\n","firstHeading":"Split View","frontmatter":null,"modified":false}'
  ]
]
[
  [
    '/Users/hendrik/Documents/Zettlr Tutorial/welcome.md',
    '{"root":false,"dir":"/Users/hendrik/Documents/Zettlr Tutorial","path":"/Users/hendrik/Documents/Zettlr Tutorial/welcome.md","name":"welcome.md","ext":".md","size":9723,"id":"","tags":[],"links":["zettelkasten"],"bom":"","type":"file","wordCount":1475,"charCount":7070,"modtime":1679442158540,"creationtime":1679442158540,"linefeed":"\\n","firstHeading":"Welcome to Zettlr!","yamlTitle":"Welcome to Zettlr!","frontmatter":{"title":"Welcome to Zettlr!","keywords":["Guide","Tutorial","Introduction"]},"modified":false}'
  ]
]
```

As you can see, the descriptors are still stringified instead of in an Object notation.

### Reproducing

Maybe it is time to retire the FSAL cache? I remember back when I implemented it I did so because I feared that the cache may otherwise become way too large, but by now – given that I by now know that the overall amount of cached data from the descriptors amounts to just under 10 MB, I think this was an overly cautious thing that now broke.

I personally don't see too much of an issue in re-doing the FSAL cache as a persistent data container. What do @kyaso and @tobiasdiez think about this? Should I fix the current `fsal-cache` implementation, or just remove it and switch to persistent data container?

### Zettlr Version

Right off the develop branch

### Specify version

v3.0.0-beta.3

### Installation Method

None

### Your Platform

- [ ] Windows
- [X] macOS
- [ ] Linux

### Architecture

- [ ] Intel 64bit
- [X] ARM 64bit

### Operating System Version

macOS Ventura 13.2.1

**Comment by nathanlesage** [comment]
Please see the referenced commit a43a431 that includes a monkey patch to make it work, but I don't know what happened during beta.2 and beta.3, to be honest.

**Comment by nathanlesage** [comment]
Quick update: I have now found the cause of this “bug”: Both the code file parser and the regular file parser have provided the cache objects as strings already; as I was calling `JSON.stringify` on the file descriptor objects, but never called `JSON.parse` afterward. I obviously just forgot that, and it's definitely embarrassing that I haven't looked into the issue for almost a year (the issue was opened in April last year).

I have now changed the parsers so that they instead run a `structuredClone` on the descriptors. In addition, I have just to be safe added a guard to loading the shards that should prevent faulty shards to be loaded from disk.
